pyquil/paulis.py

```diff
@@ -286,14 +294,7 @@ def __rsub__(self, other):
         return other + -1. * self
 
     def __str__(self):
-        term_strs = []
-        for index in self._ops.keys():
-            term_strs.append("%s%s" % (self[index], index))
-
-        if len(term_strs) == 0:
-            term_strs.append("I")
-        out = "%s*%s" % (self.coefficient, '*'.join(term_strs))
-        return out
+        return f'{self.coefficient}*{self.id(sort_ops=False)}'
```

Comment:
With this, if we do `print (sI(0))`, we'd get the awkward looking `1*` (similarly, we'd get `0*` for `print (ZERO())`. Unless this is the desired output (in the specific case of the identity operator), we could change this to `return f'{self.coefficient}*{self.id(sort_ops=False) or "I"}'

Reply:
fixed this